File: blog-backend/src/main/java/com/fonsi13/blogbackend/models/Post.java
package com.fonsi13.blogbackend.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "posts")
public class Post {

    @Id
    private String id;

    private String title;
    private String slug;
    private String content;
    private String summary;

    // Renombramos mediaUrl a coverImage para ser explícitos
    private String coverImage;

    // Lista para una galería de imágenes extra
    private List<String> images;

    // Lista de links de video (YouTube/Vimeo)
    private List<String> videoUrls;

    private String authorId;

    private List<String> topics;

    private PostStatus status;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

}

File: blog-frontend/src/types/Post.ts
export interface Post {
    id: string,
    title: string,
    slug: string,
    content: string,
    summary: string,
    coverImage: string | null,
    images: string[] | null,
    videoUrls: string[] | null,
    authorId: string,
    topics: string[],
    status: PostStatus,
    createdAt: string,

}

export type PostStatus = "DRAFT" | "PUBLISHED" | "ARCHIVED"

export interface PaginatedResponse<T>{
    content: T[],
    totalElements: number,
    totalPages: number,
    number: number,
    size: number,
    first: boolean,
    last: boolean

}